data/src/main/java/pl/edu/agh/papaya/model/UserRole.java

```diff
@@ -1,5 +1,27 @@
 package pl.edu.agh.papaya.model;
 
+import java.util.Optional;
+import java.util.stream.Stream;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import pl.edu.agh.papaya.api.model.UserStatus;
+
+@NoArgsConstructor
+@AllArgsConstructor
 public enum UserRole {
-    ADMIN,
+    ADMIN(UserStatus.ADMIN),
+    MEMBER(UserStatus.MEMBER),
+    DELETED;
```

Comment:
Ja myślałam o czymś takim. Można kwestię porządniejszego konwertera zostawić na później i na razie zostawić coś takiego albo można wywalić jakiekolwiek ślady zależności między `UserRole` i `UserStatus`, ale to trochę smutne

Btw, nazwy chyba im się należą odwrotne. W api `UserRole`, a na backendzie `UserStatus`?

Reply:
Moim zdaniem nie powinno być takiej zależności w ogóle. 

Skoro mamy problem z nazewnictwem, to pewnie chcemy mieć za dużo w jednym miejscu. Może na backendzie `RoleInProject` (albo `UserRole`) + flaga mówiąca o usunięciu użytkownika z projektu (że jest jakiś `inactive`)? 

Na froncie to pewnie zależy od faktycznych sposobów wykorzystania. Bo w ustawieniach projektu wybieramy wszystkich aktywnych userów wraz z ich uprawnieniami, a w historycznym excelu wybieramy wszystkich userów RIGH JOIN wypełnione dostępności w danym sprincie (więc tam nie potrzeba uprawnień, tylko informacja, czy user jest `inactive`, dlatego można tam w ogóle zwracać inny obiekt)

Reply:
Okeej. Ja bym zostawiła w tym enumie na backendzie `DELETED`, ale może być też bool, jeśli wszyscy będą wtedy szczęśliwsi

Reply:
nie no, ja tylko proponuję z tym podziałem

Reply:
a brak zależności wymuszam muahahah

Reply:
Ja się też nie próbuję upierać przy swoim za bardzo. Jak dla mnie `DELETED` ma sens w enumie, bo jak się dodaje kogoś usuniętego, to i tak specyfikuje się jego rolę od nowa, także trzymanie `inactive = true` oraz `userRole = UserRole.ADMIN` absolutnie nic nam nie daje

Brak zależności jest trochę smutny, bo ona tam trochę jednak jest, ale ogarniam kwestię poziomów abstrakcji - tylko będę teraz musiała zrobić issue na ten konwerter i trzeba będzie wymyślić, do czego to dopiąć, a ja już mam dosyć Jiryy

Reply:
konwerter się zrobi podczas implementacji getUsers

Reply:
w sensie to będzie taki refactor tej metody - bo przecież wszyscy robimy refactor tego co napisaliśmy, nieprawdaż?

Reply:
(a zamiast enuma DELETED, może jednak INACTIVE, hmmm????? Bo to by znaczyło, że można go właśnie aktywować ponownie, poprzez ponowne dodanie)

Reply:
> w sensie to będzie taki refactor tej metody - bo przecież wszyscy robimy refactor tego co napisaliśmy, nieprawdaż?

Jakiej metody?

Reply:
> (a zamiast enuma DELETED, może jednak INACTIVE, hmmm????? Bo to by znaczyło, że można go właśnie aktywować ponownie, poprzez ponowne dodanie)

Aaaa, nazywanierzeczyjesttrudne

Reply:
> 
> 
> > w sensie to będzie taki refactor tej metody - bo przecież wszyscy robimy refactor tego co napisaliśmy, nieprawdaż?
> 
> Jakiej metody?

no w sensie jak już ktoś napisze getUsers i będzie tam jakiś for, a w tym forze konwertowanie UserInProject na UserDTO, to pewnie żeby kod był clean i tak dalej, wydzieli do tego metodę/klasę przed wrzuceniem tego na githuba, nie? xd

Reply:
> > > w sensie to będzie taki refactor tej metody - bo przecież wszyscy robimy refactor tego co napisaliśmy, nieprawdaż?
> > 
> > 
> > Jakiej metody?
> 
> no w sensie jak już ktoś napisze getUsers i będzie tam jakiś for, a w tym forze konwertowanie UserInProject na UserDTO, to pewnie żeby kod był clean i tak dalej, wydzieli do tego metodę/klasę przed wrzuceniem tego na githuba, nie? xd

Aaa, okej, zgadzam się. Ten ktoś, kto napisze ten kod, na pewno napisze ładny kod